org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/ChainCompletionProposalComputer.java

```diff
@@ -144,20 +132,20 @@ private List<CompletionItem> executeCallChainSearch() {
 		List<Chain> found = new ArrayList<>();
 		found.addAll(mainFinder.getChains());
 		found.addAll(contextFinder.getChains());
-		return buildCompletionProposals(found);
+		buildCompletionProposals(found);
 	}
 
-	private List<CompletionItem> buildCompletionProposals(final List<Chain> chains) {
-		final List<CompletionItem> proposals = new LinkedList<>();
-
+	private void buildCompletionProposals(final List<Chain> chains) {
 		for (final Chain chain : chains) {
 			try {
-				proposals.add(create(chain));
+				var completionProposal = createCompletionProposal(chain);
+				if (completionProposal != null) {
+					coll.addAdditionalProposal(completionProposal);
```

Comment:
Could `coll.accept(..)` be used here instead ? Seems it should respect any of the filtering in there ? Does anything end up excluded that you don't want ?